src/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs

```diff
@@ -30,11 +29,11 @@ public abstract class FileSystemWatcherTest : FileCleanupTestBase
         /// Watches the Changed WatcherChangeType and unblocks the returned AutoResetEvent when a
         /// Changed event is thrown by the watcher.
         /// </summary>
-        public static AutoResetEvent WatchChanged(FileSystemWatcher watcher, string[] expectedPaths = null)
+        public static (AutoResetEvent, FileSystemEventHandler) WatchChanged(FileSystemWatcher watcher, string[] expectedPaths = null)
```

Comment:
As mentioned above, if you put names on these when you access them you can use the names. Unfortunately `ExecuteAndVerifyEvents` is structured in such a way that you have to initialize without the context so you can't leverage this. Other places you can so you should add them so you aren't doing `.Item1`.